src/reduced_basis/rb_parameters.C

```diff
@@ -153,38 +225,84 @@ Real RBParameters::get_extra_value(const std::string & param_name) const
   const auto & sample_vec = libmesh_map_find(_extra_parameters, param_name);
   libmesh_error_msg_if(sample_vec.size() != 1,
     "Requesting value for extra parameter " << param_name << ", but parameter contains multiple samples.");
+  libmesh_error_msg_if(sample_vec[0].size() != 1,
+    "Requesting Real value for extra parameter " << param_name << ", but parameter contains multiple values.");
+  return sample_vec[0][0];
+}
+
+const RBParameter & RBParameters::get_extra_vector_value(const std::string & param_name) const
+{
+  // Same as get_value(param_name) but for the map of extra parameters
+  const auto & sample_vec = libmesh_map_find(_extra_parameters, param_name);
+  libmesh_error_msg_if(sample_vec.size() == 0, "Error getting value for extra parameter " << param_name);
   return sample_vec[0];
 }
 
 Real RBParameters::get_extra_value(const std::string & param_name, const Real & default_val) const
 {
   // same as get_value(param_name, default_val) but for the map of extra parameters
   auto it = _extra_parameters.find(param_name);
-  if (it == _parameters.end())
+  if (it == _extra_parameters.end())
```

Comment:
This looks like an actual bugfix, not sure how long that has been wrong...

Reply:
Looks like I introduced it in my last PR :( I probably made a mistake with cherry-picking/rebasing between the two branches.

So at least it hasn't been a bug very long...

Reply:
I've split this out into a separate commit, to make it more clear that there was a correction here.